src/components/editor-page/document-bar/revisions/revision-list.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useMemo } from 'react'
+import { RevisionListEntry } from './revision-list-entry'
+import { useAsync } from 'react-use'
+import { getAllRevisions } from '../../../../api/revisions'
+import { useApplicationState } from '../../../../hooks/common/use-application-state'
+import { ListGroup } from 'react-bootstrap'
+import { AsyncLoadingBoundary } from '../../../common/async-loading-boundary'
+
+export interface RevisionListProps {
+  selectedRevisionId?: number
+  onRevisionSelect: (selectedRevisionId: number) => void
+}
+
+/**
+ * The list of selectable revisions of the current note.
+ *
+ * @param selectedRevisionId The currently selected revision
+ * @param onRevisionSelect Callback that is executed when a list entry is selected
+ */
+export const RevisionList: React.FC<RevisionListProps> = ({ selectedRevisionId, onRevisionSelect }) => {
+  const noteIdentifier = useApplicationState((state) => state.noteDetails.primaryAddress)
+
+  const {
+    value: revisions,
+    error,
+    loading
+  } = useAsync(() => {
+    return getAllRevisions(noteIdentifier)
+  }, [noteIdentifier])
+
+  const revisionList = useMemo(() => {
+    if (loading || !revisions) {
+      return null
+    }
+    return revisions.map((revisionListEntry) => (
+      <RevisionListEntry
+        active={selectedRevisionId === revisionListEntry.id}
+        onSelect={() => onRevisionSelect(revisionListEntry.id)}
```

Comment:
`useCallback`?

Reply:
Not necessary here because the caching is already covered by the useMemo.